From 4ae83f6699474f590927d2c4770806779703d75d Mon Sep 17 00:00:00 2001
From: Yanko Kaneti <yaneti@declera.com>
Date: Mon, 10 Feb 2014 11:49:22 +0200
Subject: [PATCH 3/3] build: Force automake C linkage when building C only

Automake chooses the linker to use by examining the _SOURCES.
Unfortunately this includes all potential sources, even the ones excluded by
CONDITIONALS, so it picks CXXLINK because of extractqt.cpp.

Linking the library with CXXLINK brings with it a soname linkage with
libstdc++.
override _LINK for the library target to force automake
to use the specific linker, C(LINK) or C++(CXXLINK)

https://bugzilla.gnome.org/show_bug.cgi?id=723877
---
 libmediaart/Makefile.am | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libmediaart/Makefile.am b/libmediaart/Makefile.am
index a7879e3..8403d23 100644
--- a/libmediaart/Makefile.am
+++ b/libmediaart/Makefile.am
@@ -34,11 +34,14 @@ libmediaart_@LIBMEDIAART_API_VERSION@_la_SOURCES = \
 
 if HAVE_GDKPIXBUF
 libmediaart_@LIBMEDIAART_API_VERSION@_la_SOURCES += extractpixbuf.c
+libmediaart_@LIBMEDIAART_API_VERSION@_la_LINK = $(LINK) $(libmediaart_@LIBMEDIAART_API_VERSION@_la_LDFLAGS)
 else
 if HAVE_QT
 libmediaart_@LIBMEDIAART_API_VERSION@_la_SOURCES += extractqt.cpp
+libmediaart_@LIBMEDIAART_API_VERSION@_la_LINK = $(CXXLINK) $(libmediaart_@LIBMEDIAART_API_VERSION@_la_LDFLAGS)
 else
 libmediaart_@LIBMEDIAART_API_VERSION@_la_SOURCES += extractdummy.c
+libmediaart_@LIBMEDIAART_API_VERSION@_la_LINK = $(LINK) $(libmediaart_@LIBMEDIAART_API_VERSION@_la_LDFLAGS)
 endif
 endif
 
-- 
1.8.5.3

